scmail.py

```diff
@@ -0,0 +1,205 @@
+    ctx.obj['pp'].pprint(key.__dict__)
+
+
+@click.command()
+@click.option('--show-private', prompt='Do you want to show the private keys?', default=False, is_flag=True,
+              help='Whether show the private keys')
+@click.pass_context
+def list_keys(ctx, show_private):
+    """List and print all the keys in the gnupg home dir."""
+    keys = ctx.obj['gpg'].list_keys(show_private)
+    key_type = 'public' if show_private is False else 'private'
+    print('You have {} {} keys.'.format(len(keys), key_type))
+    ctx.obj['pp'].pprint(keys.__dict__)
+
+
+@click.command()
+@click.pass_context
+def register(ctx):
+    """Register sc_mail API."""
+    # Request register.
+    fingerprint = ctx.obj['gpg'].list_keys(True).curkey.get('fingerprint')
+    data = {'fingerprint': fingerprint}
+
+    # Register
+    address = SECUREMAILBOX_URL + '/register/'
+    r = requests.post(address)
+    print(r.text)
+
+    if 'success' in r:
+        print('Registration success.')
+    else:
+        print('Registration fail.\nError is: {}'.format(r.get("error")))
+
+
+@click.command()
+@click.option('--recipient', prompt='The recipient', required=True, help='The email address of the recipient.')
+@click.password_option('--password', prompt='Enter passwordo of private key to decrypt', help='The passphrase of private key')
+@click.pass_context
+def retrieve(ctx, recipient, password):
+    """Retrieve and Post messages from API"""
+    # Retrieve
+    address = SECUREMAILBOX_URL + '/retrieve/'
+    r = requests.post(address)
+    # Get response
+    res = json.loads(r)
+    # print(res)
+    if 'success' in res:
+        print('The message retrieve successful.')
+    else:
+        print('The message retrieve fail.\nError is: {}'.format(res.get("error")))
+        # return
+
+    # Decrypt the messages.
+    ok, messages = ctx.obj['gpg'].decrypt_message(messages=res, passphrase=password)
+    ctx.obj['pp'].pprint(messages[:-1])
+    if ok is False:
+        print('Other message decrypt fail.\n{}'.format(messages[-1]))
+    else:
+        ctx.obj['pp'].pprint(messages[-1])
+        print('Decrypt message successful.')
+
+
+@click.command()
+@click.option('--recipient', prompt='The recipient email', required=True, help='The email of the recipient.')
+@click.option('--message', required=True, prompt='The message', help='Message that will send.')
+@click.pass_context
+def send(ctx, recipient, message):
+    """Send a message.
+
+    recipient: the fingerprint of recipient
+
+    message: the messages.
+    """
+    # Choose recipient and Encrypted
+    ok, encrypted_message = ctx.obj['gpg'].encrypt_message(message, recipient)
+    if ok is False:
+        print('Message encrypt fail.\n{}'.format(encrypted_message))
+    else:
+        print('Message encrypt success.')
+
+    # Send
+    payload = {'fingerprint': recipient, 'message': message}
+    address = SECUREMAILBOX_URL + '/send/'
+    r = requests.post(address, json=payload)
```

Comment:
Instead of importing and using `json` you could instead leverage the `.json()` method available to a requests object.
```python
r = requests.post(address, json=payload)
r = r.json()
```

Reply:
Thanks, I will change that.

Reply:
@nvenetucci just mention you because of a writer.